From f4d71e0de885010494b8b0b8d62ca910011d7544 Mon Sep 17 00:00:00 2001
From: "Max A. Dednev" <dednev@rambler.ru>
Date: Sun, 11 Jan 2015 20:00:44 -0500
Subject: [PATCH 18/18] authority: Fix memory leak in EnumerateActions call
 results handler

Policykit-1 doesn't release reference counters of GVariant data for
org.freedesktop.PolicyKit1.Authority.EnumerateActions dbus call.  This
patch fixed reference counting and following memory leak.

https://bugs.freedesktop.org/show_bug.cgi?id=88288
---
 src/polkit/polkitauthority.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/polkit/polkitauthority.c b/src/polkit/polkitauthority.c
index 75619ab..ab6d3cd 100644
--- a/src/polkit/polkitauthority.c
+++ b/src/polkit/polkitauthority.c
@@ -715,7 +715,6 @@ polkit_authority_enumerate_actions_finish (PolkitAuthority *authority,
   while ((child = g_variant_iter_next_value (&iter)) != NULL)
     {
       ret = g_list_prepend (ret, polkit_action_description_new_for_gvariant (child));
-      g_variant_ref_sink (child);
       g_variant_unref (child);
     }
   ret = g_list_reverse (ret);
-- 
2.1.0

